src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationSymbolFactory.cs

```diff
@@ -295,14 +295,15 @@ public static ITypeParameterSymbol CreateTypeParameterSymbol(string name, int or
             ImmutableArray<AttributeData> attributes,
             VarianceKind varianceKind, string name,
             ImmutableArray<ITypeSymbol> constraintTypes,
+            NullableAnnotation nullableAnnotation = NullableAnnotation.None,
```

Comment:
💭 Are we sure we want to make this optional? If it is required, it should probably be reordered with respect to `constraintTypes`.

Reply:
I think optional makes sense here. NRT is still pretty new, and None will still be the most common. Alternatively, making it required should be addressed as a separate change and be thought through all of our Create* for symbols.

Reply:
So my intuition matched @ryzngard's but looking more closely it looks like the better pattern here might be to change this overload entirely -- the only callers are taking an _existing_ TypeParameter and trying to change a few things about it.

Reply:
Ah, so yeah, the existing callers of this are generating type parameters for _declarations_ where the ? isn't allowed, which actually motivates the allowing of a default annotation of "none". But if we have a caller that also wants to specify a use that's something this should eventually support. Going to leave this as is.